src/scanspec/core.py

```diff
@@ -37,140 +31,159 @@
     "squash_frames",
     "Path",
     "Midpoints",
+    "discriminated_union_of_subclasses",
+    "StrictConfig",
 ]
 
 
-def _rec_subclasses(cls: type) -> Iterator[type]:
-    for sub_cls in cls.__subclasses__():
-        yield sub_cls
-        yield from _rec_subclasses(sub_cls)
+class StrictConfig(BaseConfig):
```

Comment:
I can't think of any downsides to forbidding it, and it is closer to the previous behaviour. I think it is probably best if the definition of each scanspec is fairly formal and a user is notified early on if it is "wrong", rather than thinking there is an extra argument in `Line` for 6 months before realising there is a problem.